dashboard/app/models/sections/section.rb

```diff
@@ -282,6 +282,34 @@ def toggle_hidden_script(script, should_hide)
     end
   end
 
+  # One of the contstraints for teachers looking for discount codes is that they
+  # have a section in which 10+ students have made progress on 5+ levels in both
+  # csd2 and csd3
+  def has_sufficient_discount_code_progress?
+    return false if students.length < 10
+    csd2 = Script.get_from_cache('csd2')
+    csd3 = Script.get_from_cache('csd3')
```

Comment:
I don't really like having logic that looks for specific scripts in the Section model, but I'm not really sure if there's a better way to accomplish this.

Reply:
Agreed - this feels like logic that would be more at home in the new `CircuitPlaygroundDiscountCodeApplication` model - I don't really want `Section` knowing anything about discount codes.  Maybe this is a good temporary place for it though.

Reply:
I think you're right that this will likely fit in a `CircuitPlaygroundDiscountCodeApplication` model.